lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -133,14 +133,10 @@ export class AuthCache extends BrowserStorage {// Singleton
         let key: string;
         // check state and remove associated cache
         for (key in storage) {
-            if (!state || key.indexOf(state) !== -1) {
-                const splitKey = key.split(Constants.resourceDelimiter);
-                const keyState = splitKey.length > 1 ? splitKey[splitKey.length-1]: null;
-                if (keyState === state && !this.tokenRenewalInProgress(keyState)) {
-                    this.removeItem(key);
-                    this.setItemCookie(key, "", -1);
-                    this.clearMsalCookie(state);
-                }
+            if ((!state || key.indexOf(state) !== -1) && !this.tokenRenewalInProgress(state)) {
```

Comment:
Did we test end-to-end? Will this pass for `concurrent` silent calls failing (at the msal-core execution) for any reason?

Reply:
Yes, I tested with the React sample. Can you clarify the concern with failing concurrent requests?

Reply:
This isn't a concern, I was thinking if there will be any cache based on state we do not want to get rid off, that should not be the case.